lib/dice/src/handoff.rs

```diff
@@ -0,0 +1,90 @@
+const ALIAS_START: usize = MEM_START;
+
+/// The Handoff type is a thin wrapper over the memory region used to transfer
+/// DICE artifacts (seeds & certs) from stage0 to hubris tasks. It is intended
+/// for use by stage0 to write these artifacts to memory where they will later
+/// be read out by a hubris task.
+pub struct Handoff<'a>(&'a RegisterBlock);
+
+impl<'a> Handoff<'a> {
+    pub fn turn_on(syscon: &'a RegisterBlock) -> Self {
+        // handoff through USB SRAM requires we power it on
+        syscon.ahbclkctrl2.modify(|_, w| w.usb1_ram().enable());
+        syscon
+            .presetctrl2
+            .modify(|_, w| w.usb1_ram_rst().released());
+
+        Self(syscon)
+    }
+
+    pub fn turn_off(self) {
+        self.0
+            .presetctrl2
+            .modify(|_, w| w.usb1_ram_rst().asserted());
+        self.0.ahbclkctrl2.modify(|_, w| w.usb1_ram().disable());
+    }
+
+    pub fn store(&self, alias_data: &AliasData) -> usize {
+        let dst_ptr = ALIAS_START as *mut [u8; AliasData::MAX_SIZE];
+        // SAFETY: Dereferencing this raw pointer is necessary to write to the
+        // memory region used to handoff DICE artifacts to Hubris tasks. This
+        // pointer will references a valid memory region provided two
+        // conditions are met:
+        // 1) The associated memory region has been enabled / turned on if
+        // necessary. This happens in the constructor / 'turn_on' function.
+        // 2) The function call is made by code in stage0.
+        let dst: &mut [u8] = unsafe { &mut *dst_ptr };
+
+        // TODO: error handling
+        hubpack::serialize(dst, alias_data).expect("serialize alias-handoff")
+    }
+}
+
+/// Type to represent DICE derived artifacts used by the root of trust for
+/// reporting in the attestation process. Stage0 will construct an instance of
+/// this type and write it to memory using the Handoff type above. The receiving
+/// hubris task will then read an AliasHandoff out of memory using the
+/// 'from_mem' constructor in the impl block.
+// TODO: This needs to be made generic to handle an arbitrary cert chain
+// instead of individual certs.
+#[derive(Deserialize, Serialize, SerializedSize)]
+pub struct AliasData {
+    pub seed: AliasOkm,
+    pub alias_cert: AliasCert,
+    pub deviceid_cert: DeviceIdSelfCert,
+}
+
+impl AliasData {
+    pub fn from_mem() -> Self {
+        let src_ptr = ALIAS_START as *const [u8; AliasData::MAX_SIZE];
+        // SAFETY: Dereferencing this raw pointer is necessary to read from the
+        // memory region used to transfer the Alias DICE artifacts from stage0
+        // to a Hubris task. This pointer will reference a valid memory region
+        // provided two conditions are met:
+        // 1) The associated memory region has been enabled / turned on if
+        // necessary. This should be done by code in stage0.
+        // 2) The task making the call has been granted access to the memory
+        // region by the kernel.
```

Comment:
Same comment about noting that this is safe even if these conditions are violated as it will cause a fault.